src/illuminatio/illuminatio_runner.py

```diff
@@ -153,92 +176,59 @@ def run_tests_for_target(network_ns, ports, target):
+    LOGGER.info("Ran nmap with cmd %s", nm_scanner.command_line())
 
-def pod_list_contains_pod(pod, pod_list):
-    """
-    Checks whether a list of pods contains a given pod
-    """
-    LOGGER.debug("Searching for pod %s", pod)
-    if isinstance(pod, ConcreteClusterHost):
-        is_on_node = any([pod == pod_on_node for pod_on_node in pod_list])
-        if is_on_node:
-            LOGGER.debug("Pod %s in namespace %s was found", pod.name, pod.namespace)
-        else:
-            LOGGER.debug(
-                "Pod %s in namespace %s isn't on this node", pod.name, pod.namespace
-            )
-        return is_on_node
-
-    LOGGER.error("Found non-ConcreteClusterHost host in cases: %s", pod)
-    return False
+    return extract_results_from_nmap(nm_scanner, port_on_nums, target)
 
 
-def extract_results_from_nmap_xml_file(result_file, port_on_nums, target):
+def extract_results_from_nmap(nmap_res, port_on_nums, target):
     """
-    Extracts the results of an nmap scan from an xml result file into a dictionary
+    Extracts the results of an nmap scan into a dictionary
     """
-    xml = ElementTree.parse(result_file.name)
-    hosts = [h for h in xml.getroot().iter("host")]
+    hosts = nmap_res.all_hosts()
     if len(hosts) != 1:
         LOGGER.error(
-            "Fund %s a single host in nmap results but expected only one target to be probed",
+            "Found %s a single host in nmap results but expected only one target to be probed",
             len(hosts),
         )
         port_string = ",".join(port_on_nums.keys())
         return {
             port_string: {
                 "success": False,
-                "error": "Found %s hosts in nmap results, expected 1."
-                % str(len(hosts)),
+                "error": f"Found {len(hosts)} hosts in nmap results, expected 1.",
             }
         }
-    host_element = hosts[0]
-    host_names = [hn.get("name") for hn in host_element.iter("hostname")]
-    LOGGER.debug("Found names %s for target %s", host_names, target)
+
     results = {}
-    for port_element in host_element.iter("port"):
-        port = port_element.get("portid")
-        state = port_element.find("state").get("state")
-        port_with_expectation = port_on_nums[port]
-        should_be_blocked = "-" in port_with_expectation
-        was_blocked = state == "filtered"
-        results[port_with_expectation] = {}
-        results[port_with_expectation]["success"] = should_be_blocked == was_blocked
-        results[port_with_expectation]["string"] = build_result_string(
-            port, target, should_be_blocked, was_blocked
-        )
-        results[port_with_expectation]["nmap-state"] = state
+
+    host = hosts[0]
+    for proto in nmap_res[host].all_protocols():
+        # get all scanned ports for all protocols
+        for port in nmap_res[host][proto].keys():
+            if proto == "tcp":
+                state = nmap_res[host].tcp(port)["state"]
+            else:
```

Comment:
Is `nmap_res[host]["tcp"]` not usable? This condition looks weird, maybe a comment on why this is needed would be helpful.

Reply:
It's usable but I was not able to mock the command in a fast and easy way.